utils/s2n_mem.c

```diff
@@ -293,6 +293,16 @@ int s2n_free(struct s2n_blob *b)
     return S2N_SUCCESS;
 }
 
+int s2n_free(struct s2n_blob *b)
+{
+    return s2n_free_impl(b, true);
+}
```

Comment:
Rather than a boolean to a common impl, can s2n_free just call s2n_blob_zero and then s2n_free_non_zeroed? You shouldn't need an extra s2n_blob_validate if you don't touch any blob fields directly.

Reply:
The opaqueness of what true vs false means here contributed to my confusion related to https://github.com/aws/s2n-tls/pull/3472#discussion_r958799886

Reply:
I wanted to avoid extra validation calls since this is a hot path, especially with dynamic buffers enabled. If I made a comment on the bool would that help? Or an enum?

Reply:
It shouldn't cause any extra validation calls. If you're just calling other blob methods, you don't need to re-validate. That's only required if you mess with the fields of the blob struct directly.

```
int s2n_free(struct s2n_blob *b)
{
    /* To avoid memory leaks, don't exit the function until the memory has been freed */
    int zero_rc = s2n_blob_zero(b);
    POSIX_GUARD(s2n_free_non_zeroed(b));
    return zero_rc;
}

int s2n_free_non_zeroed(struct s2n_blob *b)
{
    POSIX_PRECONDITION(s2n_blob_validate(b));

    POSIX_ENSURE(initialized, S2N_ERR_NOT_INITIALIZED);
    POSIX_ENSURE(s2n_blob_is_growable(b), S2N_ERR_FREE_STATIC_BLOB);
    POSIX_GUARD(s2n_mem_free_cb(b->data, b->allocated));
    *b = (struct s2n_blob) {0};
    POSIX_GUARD(zero_rc);

    return S2N_SUCCESS;
}
```

Reply:
Makes sense; I can do that instead.